graphqlmetrics/pkg/telemetry/config.go

```diff
@@ -0,0 +1,58 @@
+package telemetry
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"go.opentelemetry.io/otel/attribute"
+
+	otelmetric "go.opentelemetry.io/otel/metric"
+)
+
+const (
+	DefaultServerName     = "cosmo-graphqlmetrics"
+	defaultServiceVersion = "dev"
+)
+
+type CustomMetrics struct {
+	counters map[string]otelmetric.Int64Counter
+}
+
+// NewTelemetryConfig creates the config to be used for the
+// telemetry inside graphqlmetrics.
+func NewTelemetryConfig(serviceVersion string, prometheusConfig PrometheusConfig) *Config {
+	if serviceVersion == "" {
+		serviceVersion = defaultServiceVersion
+	}
+
+	return &Config{
+		Name:       DefaultServerName,
+		Version:    serviceVersion,
+		Prometheus: prometheusConfig,
+	}
+}
+
+// Config represents the configuration for the agent.
+type Config struct {
+	// Name represents the service name for metrics. The default value is cosmo-router.
+	Name string
+	// Version represents the service version for metrics. The default value is dev.
+	Version string
+	// Prometheus includes the Prometheus configuration
+	Prometheus PrometheusConfig
+	// CustomPrometheusMetrics to be collected
+	CustomMetrics CustomMetrics
```

Comment:
Is this for testing?

Reply:
I am not sure what you mean here.

I updated some comments, it's not reflected here in the PR view, I hope this makes more sense:

https://github.com/wundergraph/cosmo/commit/bd2a30ff4c577f373677a49888a398bddd66a961